### PR TITLE
[ENH] Add DummyClassifier

### LINKED ISSUE #2507: [ENH] DummyClassifier
We should implement a `DummyClassifier`, similar to the one in `sklearn`.
Functionality can be equivalent, except that it should inherit from `BaseClassifier` in `sktime`.

Suggestion: impot `sklearn` `DummyClassifier`, loop through all the inputs and loop through `y`, create some 2D `X` for the `sklearn` classifier which will be ignored anyway.

### PR DESCRIPTION
**Pull request by ZiyaoWei**
Fixes #2507.

#### What does this implement/fix? Explain your changes.

Adds a DummyClassifier that wraps over a sklearn DummyClassifier and inherits from BaseClassifier.

#### Does your contribution introduce a new dependency? If yes, which one?

No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [DOC] or [BUG] indicating whether the PR topic is related to enhancement, documentation or bug

### CODE REVIEW

**Comment by ZiyaoWei** [comment]
@fkiraly - so internally this didn't create any fake data and instead just passed the data directly to the sklearn DummyClassifier - is that okay?

Also, I guess this is the only one where sklearn has a direct equivalent, but would we want any other dummy implementations?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thank you!

Currently, your classifier does not seem to pass any tests.

You need to make sure you adhere to the `sktime` specifications. For instance, the `X` passed are 3D numpy array, or whatever is specified in the `X_inner_mtype`.

Since the `DummyClassifier` ignores `X`, you could just create an `sklearn` array of suitable 2D shape to avoid formatting issues.

Further, `__init__` should not have an args catch.

**Comment by ZiyaoWei** [comment]
@fkiraly - thanks! I didn't realize the test suite enforces these, will fix later. Sorry for the back and forth!

**Comment by ZiyaoWei** [comment]
It turns out I forgot to call the super class initializer - not a Python expert by any means!

@fkiraly - it seems now it passes the tests; however, I didn't create any dummy input, and since the input is converted by the boilerplate in `BaseClassifier` and then ignored the underlying calls to sklearn still works. Am I missing something?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks, looks great!

I would kindly request one change:
This should also work for unequal length series, but by virtue of the conversion to `numpy3D` the classifier only works for equal length series. How to deal with that, see above.

Let me know if you are not sure what I mean.

File: sktime/classification/dummy/_dummy.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        ----------
+        X : sktime-format pandas dataframe with shape(n,d),
+        or numpy ndarray with shape(n,d,m)
+        y : array-like, shape = [n_instances] - the class labels
+
+        Returns
+        -------
+        self : reference to self.
+        """
+        self.sklearn_dummy_classifier.fit(X, y)
```
[the remark this answers is not in the document]
Comment (ZiyaoWei) [reply sktime/classification/dummy/_dummy.py:46]: Thanks! I thought this has the advantage of not having to create anything so it's more efficient, but I guess that would be unnecessary optimization and creating a dummy value makes the code more readable - will change.

File: sktime/classification/dummy/_dummy.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+class DummyClassifier(BaseClassifier):
+    """DummyClassifier makes predictions that ignore the input features.
+
+    This classifier serves as a simple baseline to compare against other more
+    complex classifiers, similar to sklearn.dummy.DummyClassifier.
+    """
+
+    _tags = {
+        "capability:missing_values": True,
```
[the remark this answers is not in the document]
Comment (ZiyaoWei) [reply sktime/classification/dummy/_dummy.py:21]: I had this initially but weren't sure how to make it work, and later forgot to add it back. Thanks for the pointers!

**Comment by fkiraly** [comment]
@ZiyaoWei, let me know once ready for re-review.
There is also a conflict with `.all-contributorsrc`, kindly resolve that or I cannot trigger CI/CD

**Review by fkiraly — APPROVED** [review]
@ZiyaoWei, thanks a lot! Good to go now!

I made some minor change before merge:

* fixing the merge conflict in the contributor file
* adding a docstring, copy-paste from sklearn with credit attribution

**Comment by ZiyaoWei** [comment]
@fkiraly - thanks for the reviews and the cleanup fixes,  happy to contribute!
